src/getdata.js

```diff
@@ -0,0 +1,25 @@
+const xhr = new XMLHttpRequest();
+const url = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/6XXWCXn9VJxphtkrJMBV/scores/';
+
+export default class GetData {
+  constructor() {
+    this.scoreData = [];
+  }
+
+  async getScore() {
+    const promise = new Promise((myResolve) => {
+      xhr.open('GET', url);
+      xhr.onload = () => {
+        if (xhr.status === 200) {
+          myResolve(xhr.response);
+        } else {
+          myResolve('Error');
+        }
+      };
+      xhr.send();
+    });
+    const scoreList = JSON.parse(await promise);
+    this.scoreData = scoreList.result;
+    this.scoreData.sort((a, b) => b.score - a.score);
+  }
+}
```

Comment:
- [ ] I would recommend using `async` and `await` with the `fetch` method to send and receive data. It is fully asynchronous and Promise based and it requires less code compared to XMLHttpRequest. you can know more about both `XMLHttpRequest` and `async-await` [here](https://betterprogramming.pub/from-xhr-to-fetch-with-async-await-on-the-finish-line-b021de1d226b).

Reply:
thanks and I will read through the async and await fetch method for my reference